advanced/is-pattern-1/templates/is/wso2is-pattern-1-identity-server-bin.yaml

```diff
@@ -351,8 +350,7 @@ data:
         -Dorg.apache.xml.security.ignoreLineBreaks=false \
         -Dnashorn.args="$NASHRON_ARGS" \
         -Dcarbon.new.config.dir.path="$CARBON_HOME/repository/resources/conf" \
-        -javaagent:/home/wso2carbon/prometheus/jmx_prometheus_javaagent-0.12.0.jar=2222:/home/wso2carbon/prometheus/config.yaml \
+        {{ if .Values.wso2.monitoring.enabled }} -Djdk.tls.client.protocols=TLSv1.2 -javaagent:/home/wso2carbon/prometheus/jmx_prometheus_javaagent-0.12.0.jar=2222:/home/wso2carbon/prometheus/config.yaml \{{- else }}-Djdk.tls.client.protocols=TLSv1.2 \{{ end }}
```

Comment:
Can't we have this like following,
```
-Dnashorn.args="$NASHRON_ARGS" \
-Dcarbon.new.config.dir.path="$CARBON_HOME/repository/resources/conf" \
-Djdk.tls.client.protocols=TLSv1.2 \
{{ if .Values.wso2.monitoring.enabled }}  -javaagent:/home/wso2carbon/prometheus/jmx_prometheus_javaagent-0.12.0.jar=2222:/home/wso2carbon/prometheus/config.yaml \{{ end }}
```
instead of bringing the `-Djdk.tls.client.protocols=TLSv1.2 \` part into the logic where it is a constant.

Reply:
@ThilinaManamgoda I tried it. The issue is that when `.Values.wso2.monitoring.enabled` is disabled, this approach will generate an empty new line, when rendering the ConfigMap template instead of just ignoring the line.

```
    -Dnashorn.args="$NASHRON_ARGS" \
    -Dcarbon.new.config.dir.path="$CARBON_HOME/repository/resources/conf" \
    -Djdk.tls.client.protocols=TLSv1.2 \

    -Dcarbon.new.config.dir.path="$CARBON_HOME/repository/resources/conf" \
    org.wso2.carbon.bootstrap.Bootstrap $*
```

This leads to an erroneous server startup with failures in the JVM (as per my understanding due to the newline).

Hence, I had to follow this approach.

Reply:
@ThilinaManamgoda the following is the error.

```
JAVA_HOME environment variable is set to /usr/local/java/jdk1.8.0_131
CARBON_HOME environment variable is set to /home/chiranga/Desktop/wso2is-5.10.0
Using Java memory options: -Xms256m -Xmx1024m
Usage: java [-options] class [args...]
           (to execute a class)
   or  java [-options] -jar jarfile [args...]
           (to execute a jar file)
where options include:
    -d32	  use a 32-bit data model if available
    -d64	  use a 64-bit data model if available
    -server	  to select the "server" VM
                  The default VM is server,
                  because you are running on a server-class machine.


    -cp <class search path of directories and zip/jar files>
    -classpath <class search path of directories and zip/jar files>
                  A : separated list of directories, JAR archives,
                  and ZIP archives to search for class files.
    -D<name>=<value>
                  set a system property
    -verbose:[class|gc|jni]
                  enable verbose output
    -version      print product version and exit
    -version:<value>
                  Warning: this feature is deprecated and will be removed
                  in a future release.
                  require the specified version to run
    -showversion  print product version and continue
    -jre-restrict-search | -no-jre-restrict-search
                  Warning: this feature is deprecated and will be removed
                  in a future release.
                  include/exclude user private JREs in the version search
    -? -help      print this help message
    -X            print help on non-standard options
    -ea[:<packagename>...|:<classname>]
    -enableassertions[:<packagename>...|:<classname>]
                  enable assertions with specified granularity
    -da[:<packagename>...|:<classname>]
    -disableassertions[:<packagename>...|:<classname>]
                  disable assertions with specified granularity
    -esa | -enablesystemassertions
                  enable system assertions
    -dsa | -disablesystemassertions
                  disable system assertions
    -agentlib:<libname>[=<options>]
                  load native agent library <libname>, e.g. -agentlib:hprof
                  see also, -agentlib:jdwp=help and -agentlib:hprof=help
    -agentpath:<pathname>[=<options>]
                  load native agent library by full pathname
    -javaagent:<jarpath>[=<options>]
                  load Java programming language agent, see java.lang.instrument
    -splash:<imagepath>
                  show splash screen with specified image
See http://www.oracle.com/technetwork/java/javase/documentation/index.html for more details.
./bin/wso2server.sh: 332: ./bin/wso2server.sh: -Dcarbon.new.config.dir.path=/home/chiranga/Desktop/wso2is-5.10.0/repository/resources/conf: not found
```
We can fix this using the following approach.

```
 -Dorg.apache.xml.security.ignoreLineBreaks=false \
 -Dnashorn.args="$NASHRON_ARGS" \
    \
    -Dcarbon.new.config.dir.path="$CARBON_HOME/repository/resources/conf" \
```
But, IMO the above logic will eliminate the need for an unnecessary newline.

Please feel free to drop in your thoughts and concerns.

Reply:
@chirangaalwis Could you please try the following syntax,
```
 {{- if .Values.wso2.monitoring.enabled }}-javaagent:/home/wso2carbon/prometheus/jmx_prometheus_javaagent-0.12.0.jar=2222:/home/wso2carbon/prometheus/config.yaml \{{- end }}
```

Reply:
@ThilinaManamgoda yes it seems that the missing `-` in `{{- if...` leads to a new line as follows.
![Screenshot from 2020-08-20 00-30-25](https://user-images.githubusercontent.com/9444555/90678970-189db700-e27d-11ea-995d-bb782ffea25c.png)

Please find the fixed resource.